demo/docker-compose.yml

```
@@ -1,36 +1,42 @@
-version: '3.7'
+version: '3.8'
+volumes:
+  pgdata:
+    driver: local
 services:
   postgres:
-    build:
-      context: postgres
-      dockerfile: Dockerfile
-    shm_size: 256MB
+    image: postgres:15beta2
+    # https://dba.stackexchange.com/q/275378/46472
+    shm_size: 3GB
     volumes:
+      - type: volume
+        source: pgdata
+        target: /var/lib/postgresql/data/
       - type: bind
-        source: ./postgres/db
-        target: /db
-    sysctls:
-      net.core.somaxconn: 256
-      net.ipv4.tcp_syncookies: 0
-      # To use all of 2 gigabytes
-      kernel.shmmax: 1055092736
-      kernel.shmall: 257591
-    privileged: true
+        source: ./postgres/init
+        target: /docker-entrypoint-initdb.d
     environment:
-      PG_TRUST_LOCALNET: 'true'
-    command: tail -f /dev/null
+      POSTGRES_DB: fhirdb
+      POSTGRES_USER: fhiradmin
+      POSTGRES_PASSWORD: change-password
+      PGDATA: /var/lib/postgresql/data/pgdata
     healthcheck:
-      start_period: 21s
       test: ["CMD-SHELL", "pg_isready -d fhirdb -U fhiradmin"]
-      interval: 10s
-      timeout: 5s
+      interval: 5s
```

I suppose this is OK for a new database. If the database were restarting and needed to run recovery it could take longer to start up. The short setting is probably fine for a CI environment, but for a demo environment where the database may be restarted (perhaps after a crash), this might be an issue.

---

what i found:  the demo env uses a persistent volume, so after the initial setup the postgres image was routinely starting in less than 5 seconds.  however, because our fhir service delays its startup until the db is healthy, we were waiting needlessly.
however, we can add a `start_period` back in to ensure that an initial slow startup doesn't cause use any trouble (while still allowing it to become "healthy" in 5 seconds if its ready).
```suggestion
      start_period: 22s
      interval: 5s
```